### DISCUSSION TITLE
How should I use tar_load() from a .Rmd in a subfolder?

### DISCUSSION THREAD

**Discussion by JohannesNE** [ropensci/targets#781] [Help] [answered]
I use targets to process my data. I then want to load some targets in report files (.Rmd), but (for now) I do not want the reports as targets.

If I make a file: `reports/results.Rmd` that looks like this:

````
```{r}
targets::tar_load(subjects)
```
````

it fails:

```
Error in `tar_throw_validate()`:
! missing files: _targets/meta/meta
Backtrace:
 1. targets::tar_load(subjects)
 3. targets::tar_meta(targets_only = TRUE, store = store)
 4. targets::tar_assert_path(path_meta(path_store = store))
 5. targets::tar_throw_validate(...)
```

I think this is because it does not find my `_targets.yaml` with a custom store path.

I can make it work by:
  1. Moving  `reports/results.Rmd` to the project root. (adding `knitr::opts_chunk$set(echo = TRUE, root.dir = here::here())` did not help).
  2. Explicitly specifying the target store path:
````
```{r}
tar_store_ <- here::here("data/Intermediate_data/_targets")
tar_meta_ <- tar_meta(store = tar_store_)
tar_load(subjects, meta=tar_meta_, store = tar_store_)
````

Is there a way to apply the `_targets.yaml` from the `.Rmd` file, or is there another recommended way to solve this problem?

**Comment by ginolhac** [comment, marked as answer]
Hello,

I have a similar setup and when the Rmd is not at the project root, I am doing something like this:

``` r
config <- withr::with_dir(rprojroot::find_root('_targets.R'), tar_read(conf))
```

For the knitting, you also specify the root with the same thing:

``` r
knitr::opts_chunk$set(root.dir = rprojroot::find_root('_targets.R'))
```

Hope this helps

**Reply by JohannesNE** [reply]
Great. That works! I just use `here::here()` instead of `rprojroot::find_root('_targets.R')`.

I still do not understand why it is not enough to just set the `root.dir`.

**Reply by ginolhac** [reply]
`here` is a wrapper (useless if you ask me) of `rprojroot`. 
There is some parts of the `callr` calls by {targets} that I don't understand, I only know the workarounds. 
Like for `render_site()` that I wrapped inside a `callr` sub-call.
